### DISCUSSION TITLE
Permission descriptions

### DISCUSSION THREAD

**Discussion by runthis** [spatie/laravel-permission#1827] [Ideas]
Hi!

Very convenient tool, thank you!

I had a feature idea for being able to provide a description to a permission when creating one. I know you can make permission names verbose, but being able to add a small, maybe like 255 character limit description could be nice.

Thanks!

**Comment by runthis** [comment]
If anyone wants to implement this themselves in the future, I added a migration to add a "description" column to the permissions table, then just tossed in a "description" key when creating a permission.

migration:
```
$table->string('description')->default('');
```

Creating a permission

```
Permission::create([
    'name' => 'do:something',
    'description' => 'Does something that is something.'
]);
```

**Reply by erikn69** [reply]
yes that's the way to achieve that, you can add all the fields you need
